src/cmds/pbsdsh.c

```diff
@@ -95,6 +107,96 @@ bailout(int sig)
  * @return - Void
  *
  */
+
+
+int
+host_match(char *line, char *host)
+{
+	if (NULL == host ) return 0;
+	int len = strlen(line);
+	static char domain[PBS_MAXHOSTNAME + 1];
+	char fullhost[PBS_MAXHOSTNAME + 1];
+	static struct in_addr addr;
+	static int addrvalid = -1;
+
+	if (line[len - 1] == '\n')
+		line[len - 1] = '\0';
+
+	if (strcmp(line, host) == 0)
+		return 1;
+
+	if (addrvalid == -1) {
+		addr.s_addr = inet_addr(host);
+		addrvalid = (addr.s_addr == INADDR_NONE) ? 0 : 1;
+	}
+
+	if (addrvalid) { /* compare IP addresses */
+		struct hostent *hp = gethostbyname(line);
+		int i;
+
+		if (hp == NULL)
+			return 0;
+
+		for (i = 0; hp->h_addr_list[i]; i++) {
+			if (memcmp(&addr, hp->h_addr_list[i],
```

Comment:
hostent->h_addr_list() is a char**.  Is there a reason you chose to use memcmp()?

Excuse my ignorance, but I'm a little confused.  According to the man page, h_length is the length of the "address" in bytes.  Then h_addr_list is an array of network addresses.  Can the length of each of the addresses differ?

Reply:
I dont know either, :).  The problem is resolved after changing to getaddrinfo.